src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    public static final String DELIMITER = ",";
+    public static final String OPERATION = "buy";
+    private int supplyAmount = 0;
+    private int buyAmount = 0;
+    private final StringBuilder stringBuilder = new StringBuilder();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        supplyAmount = 0;
+        buyAmount = 0;
+        File fromFile = new File(fromFileName);
+        final File toFile = new File(toFileName);
+
+        readDataFromTheFile(fromFile);
+
+        int result = supplyAmount - buyAmount;
+
+        stringBuilder.setLength(0);
+        stringBuilder.append("supply,").append(supplyAmount).append(System.lineSeparator())
+                .append("buy,").append(buyAmount).append(System.lineSeparator())
+                .append("result,").append(result);
+
+        writeToTheFile(toFile);
+    }
+
+    private void readDataFromTheFile(File fromFile) {
+        List<String> fileData;
+        try {
+            fileData = Files.readAllLines(fromFile.toPath());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file", e);
+        }
+
+        for (String data : fileData) {
+            String[] values = data.split(DELIMITER);
+
+            if (values.length != 2) {
+                continue;
+            }
+
+            String operation = values[0];
+            String amount = values[1];
+
+            try {
+                int value = Integer.parseInt(amount);
+                if (operation.equals(OPERATION)) {
+                    buyAmount += value;
+                } else {
+                    supplyAmount += value;
+                }
+            } catch (NumberFormatException e) {
+                System.err.println("Invalid number format in file: " + amount);
```

Comment:
Ignoring the `NumberFormatException` by just printing to `System.err` is not ideal. It would be better to throw a runtime exception with a message that includes the file name and the invalid data that caused the exception.